framework/src/meshgenerators/Poly2TriMeshGenerator.C

```diff
@@ -0,0 +1,242 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "Poly2TriMeshGenerator.h"
+
+#include "CastUniquePointer.h"
+#include "MooseUtils.h"
+
+#include "libmesh/elem.h"
+#include "libmesh/int_range.h"
+#include "libmesh/mesh_triangle_holes.h"
+#include "libmesh/parsed_function.h"
+#include "libmesh/poly2tri_triangulator.h"
+#include "libmesh/unstructured_mesh.h"
+
+registerMooseObject("MooseApp", Poly2TriMeshGenerator);
+
+InputParameters
+Poly2TriMeshGenerator::validParams()
+{
+  InputParameters params = MeshGenerator::validParams();
+
+  MooseEnum algorithm("BINARY EXHAUSTIVE", "BINARY");
+
+  params.addRequiredParam<MeshGeneratorName>(
+      "boundary", "The MeshGenerator that defines the mesh outer boundary.");
+  params.addParam<unsigned int>(
+      "interpolate_boundary", 0, "How many more nodes to add in each outer boundary segment.");
+  params.addParam<bool>(
+      "refine_boundary", true, "Whether to allow automatically refining the outer boundary.");
+  params.addParam<bool>("smooth_triangulation",
+                        false,
+                        "Whether to do Laplacian mesh smoothing on the generated triangles.");
+  params.addParam<std::vector<MeshGeneratorName>>(
+      "holes", std::vector<MeshGeneratorName>(), "The MeshGenerators that define mesh holes.");
+  params.addParam<std::vector<bool>>(
+      "stitch_holes", std::vector<bool>(), "Whether to stitch to the mesh defining each hole.");
+  params.addParam<std::vector<unsigned int>>("interpolate_holes",
+                                             std::vector<unsigned int>(),
+                                             "How many nodes to add per hole boundary segment.");
+  params.addParam<std::vector<bool>>("refine_holes",
+                                     std::vector<bool>(),
+                                     "Whether to allow automatically refining each hole boundary.");
+  params.addParam<Real>(
+      "desired_area", std::numeric_limits<Real>::max(), "Desired (maximum) triangle area.");
+  params.addParam<std::string>("desired_area_func",
+                               std::string(),
```

Comment:
```suggestion
  params.addParam<FunctionName>("desired_area_func",
                               std::string(),
```
does this work

Reply:
oh this function is a parsed expression not a moose function?

Reply:
Correct; it goes into a libMesh API expecting an fparser friendly string.